src/dawn/engine_dawn.cc

```diff
@@ -700,41 +789,79 @@ Result DawnPipelineHelper::CreateRenderPipelineDescriptor(
           "the render pipeline");
     }
   }
-
-  // Set defaults for the vertex stage descriptor.
-  vertexStage.module = render_pipeline.vertex_shader;
-  vertexStage.entryPoint = vertexEntryPoint.c_str();
-  renderPipelineDescriptor.vertexStage = &vertexStage;
-
-  // Set defaults for the fragment stage descriptor.
-  fragmentStage.module = render_pipeline.fragment_shader;
-  fragmentStage.entryPoint = fragmentEntryPoint.c_str();
-  renderPipelineDescriptor.fragmentStage = std::move(&fragmentStage);
-
-  // Set defaults for the input state descriptors.
-  tempInputState.indexFormat = ::dawn::IndexFormat::Uint32;
   // Fill the default values for vertexInput.
-  vertexInput.inputSlot = 0;
-  vertexInput.stride = 0;
-  vertexInput.stepMode = ::dawn::InputStepMode::Vertex;
+  // assuming #inputs = #attributes
+  // if isDrawRectCommand == true ignore index/vertex buffers that are are
+  // attached to the pipeline and attach one buffer with format of Float4  and
+  // 4*sizeof(float) stride (format and stride of the buffer that we create in
+  // DoDrawRect)
+  int num_inputs = 0;
   for (uint32_t i = 0; i < kMaxVertexInputs; ++i) {
+    if (isDrawRectCommand && i == 0) {
+      vertexInput.inputSlot = i;
+      vertexInput.stride = 4 * sizeof(float);
+      vertexInput.stepMode = ::dawn::InputStepMode::Vertex;
```

Comment:
Ah.  I see that AmberScript does not support instanced drawing.  But vkscript does have a syntax for instanced drawing.
I filed #565 to support it  in AmberScript.  Also #566 for the Dawn engine support, which affects this line.

Sorry for the misdirection earlier on this.